spec/models/info_request_spec.rb

```diff
@@ -1682,6 +1682,32 @@
       it { is_expected.to include(guess) }
     end
 
+    context 'upper case email with a broken id and otherwise intact idhash' do
+      let(:email) { "REQUEST-123a-#{ info_request.idhash.upcase }@example.com" }
+      let(:guess) { described_class::Guess.new(info_request, email, :idhash) }
+      it { is_expected.to include(guess) }
+    end
+
+    context 'correct id and idhash with no punctuation' do
+      let(:email) do
+        "request#{ info_request.id }#{ info_request.idhash }@example.com"
+      end
+
+      it 'includes a guess related to the expected request' do
+        expect(subject[0].info_request).to eq info_request
```

Comment:
Transient spec fail avoidance. As the email matches on both the id and the hash, we're going to generate 2 `Guess`es and filter out one of them but we can't be certain which will be returned.

Reply:
I can't prove this. Logically it _should_ always pick the `Guess` instance that matches on `:id` but ... I was sure I'd seen it occasionally fail because it somehow ended up with the `:idhash` instance instead.

But I've just run the code on loop (3 sets of 200,000 reps) without it returning the `:idhash` instance so ... maybe this precaution is not required after all? (I'm inclined to rewrite this and describe the potential effects in the PR notes instead.)